File: map/src/api/model/page.ts
type Page<T> = {
    content: T[],
    page: {
        size: number,
        number: number,
        totalElements: number,
        totalPages: number,
    },
}

type PageRequest = {
    page: number,
    size: number,
}

export type { Page, PageRequest }

File: map-api/src/main/kotlin/domain/model/Page.kt
package cz.cvut.fit.gaierda1.domain.model

data class Page<T>(
    val content: List<T>,
    val totalPages: Int,
    val totalElements: Long,
    val pageRequest: PageRequest,
) {
    fun <S> map(transform: (T) -> S) = Page(
        content = content.map(transform),
        totalPages = totalPages,
        totalElements = totalElements,
        pageRequest = pageRequest
    )
}
